utilities/rpp-unittests/HIP/Single_hip.cpp

```diff
@@ -0,0 +1,607 @@
+#include <stdio.h> 
+#include <dirent.h>
+#include <string.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include "/opt/rocm/rpp/include/rppi.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <time.h>
+#include "hip/hip_runtime_api.h"
+using namespace cv;
+using namespace std;
+#define images 100
+
+void check_hip_error(void)
+{
+	hipError_t err = hipGetLastError();
+	if (err != hipSuccess)
+	{
+ 		cerr<< "Error: "<< hipGetErrorString(err)<<endl;
+		exit(err);
+	}
+}
+int G_IP_CHANNEL = 3;
+int G_MODE = 2;
+char funcType[1000] = {"Single"};
+char src[1000] = {"/home/ulagammai/ulagammai/TESTSUITE_RPP/Input_Images/RGB"};
+char src_second[1000] = {"/home/ulagammai/ulagammai/TESTSUITE_RPP/Input_Images/RGBS1"};
+char dst[1000] = {"/home/ulagammai/Hip-Unit-Testing/output"};
+
+int main(int argc, char **argv)
+{
+	int ip_channel = G_IP_CHANNEL;
+	int mode = G_MODE;
+	char *funcName = argv[1]; 
+	//char *src = argv[2];
+	//char *dst = argv[3];
+	//char *src2 = argv[4];
+	
+	if(mode == 0)
+	{
+		strcat(funcType,"_CPU");
+	}
+	else if (mode == 1)
+	{
+		strcat(funcType,"_GPU");
+	}
+	else
+	{
+		strcat(funcType,"_HIP");
+	}
+	if(ip_channel == 1)
+	{
+		strcat(funcType,"_PLN");
+	}
+	else
+	{
+		strcat(funcType,"_PKD");
+	}
+	
+	int i = 0, j = 0;
+	int minHeight = 30000, minWidth = 30000, maxHeight = 0, maxWidth = 0;
+	
+	unsigned long long ioBufferSize = 0, oBufferSize = 0;
+	
+	static int noOfImages = 0;
+
+	Mat image,image_second;
+	
+	struct dirent *de;
+	char src1[1000];
+	strcpy(src1, src);
+	strcat(src1, "/");
+	char src1_second[1000];
+	strcpy(src1_second, src_second);
+	strcat(src1_second, "/");
+	strcat(funcName,funcType);
+	strcat(dst,"/");
+	strcat(dst,funcName);
+	mkdir(dst, 0700);
+	strcat(dst,"/");
+
+	DIR *dr = opendir(src); 
+	while ((de = readdir(dr)) != NULL) 
+	{
+		if(strcmp(de->d_name,".") == 0 || strcmp(de->d_name,"..") == 0)
+			continue;
+		noOfImages += 1;
+		break;
+	}
+	closedir(dr);
+
+	RppiSize *srcSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+	RppiSize *dstSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
```

Comment:
Codacy found an issue: [Invalid calloc() argument nr 1. The value is 0 but the valid values are '1:'.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)